wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/Components.java

```
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.devfile.server;
+
+import static org.eclipse.che.api.devfile.server.Constants.DOCKERIMAGE_COMPONENT_TYPE;
+import static org.eclipse.che.api.devfile.server.Constants.EDITOR_COMPONENT_TYPE;
+import static org.eclipse.che.api.devfile.server.Constants.KUBERNETES_COMPONENT_TYPE;
+import static org.eclipse.che.api.devfile.server.Constants.OPENSHIFT_COMPONENT_TYPE;
+import static org.eclipse.che.api.devfile.server.Constants.PLUGIN_COMPONENT_TYPE;
+
+import org.eclipse.che.api.core.model.workspace.devfile.Component;
+
+/** Utility methods for working with components */
+public class Components {
+
+  private Components() {}
+
+  /**
+   * Get a name that can be used to identify the component in the devfile. Either the component
+   * alias is used or, if not defined, the identifying attribute corresponding to the component
+   * type.
+   *
+   * @return
```

Can we reformulate summary(see [Google Style Guide](https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment)) and remove this?

---

Also there is no note that it may return null

---

I've changed it to not return null but rather throw an exception on the illegal component name.